test/acceptance/full-dump-replication.js

```diff
@@ -0,0 +1,154 @@
+  let targetDatabaseConfig;
+
+  before(async () => {
+    const rawSourceDataBaseConfig = pgUrlParser(SOURCE_DATABASE_URL);
+
+    sourceDatabaseConfig = {
+      serverUrl: `postgres://${rawSourceDataBaseConfig.user}@${rawSourceDataBaseConfig.host}:${rawSourceDataBaseConfig.port}`,
+      databaseName: rawSourceDataBaseConfig.database,
+      tableName: 'test_table',
+      tableRowCount: 100000,
+    };
+
+    sourceDatabaseConfig.databaseUrl = `${sourceDatabaseConfig.serverUrl}/${sourceDatabaseConfig.databaseName}`;
+
+    const rawTargetDataBaseConfig = pgUrlParser(TARGET_DATABASE_URL);
+
+    targetDatabaseConfig = {
+      serverUrl: `postgres://${rawSourceDataBaseConfig.user}@${rawTargetDataBaseConfig.host}:${rawTargetDataBaseConfig.port}`,
+      databaseName: rawTargetDataBaseConfig.database,
+      tableName: 'test_table',
+      tableRowCount: 100000,
+    };
+
+    targetDatabaseConfig.databaseUrl = `${targetDatabaseConfig.serverUrl}/${targetDatabaseConfig.databaseName}`;
+
+    // given
+    const configuration = {
+      SOURCE_DATABASE_URL,
+      TARGET_DATABASE_URL,
+      DATABASE_URL: TARGET_DATABASE_URL,
+      RESTORE_ANSWERS_AND_KES: 'true',
+      PG_RESTORE_JOBS: 1,
+      AIRTABLE_API_KEY: 'keyblo10ZCvCqBAJg',
+      AIRTABLE_BASE: 'app3fvsqhtHJntXaC'
+    };
+    sourceDatabase = await Database.create(sourceDatabaseConfig);
+    await createAndFillDatabase(sourceDatabase, sourceDatabaseConfig, { createTablesNotToBeImported: true });
+    targetDatabase = await Database.create(targetDatabaseConfig);
+
+    // when
+    await steps.fullReplicationAndEnrichment(configuration);
+
+  });
+
+  after(async function() {
+    await sourceDatabase.dropDatabase();
+    await targetDatabase.dropDatabase();
+  });
+
+  describe('should import database data', () => {
+
+    it('should replicate answers and knowledge-elements ', async () => {
+
+      // then
+      const restoredRowCount = parseInt(await targetDatabase.runSql(`SELECT COUNT(*) FROM ${targetDatabaseConfig.tableName}`));
+      expect(restoredRowCount).to.equal(targetDatabaseConfig.tableRowCount);
+
+      const isAnswersRestored = parseInt(await targetDatabase.runSql('SELECT  COUNT(1) FROM information_schema.tables t WHERE t.table_name = \'answers\''));
+      expect(isAnswersRestored).to.equal(1);
+
+      // then
+      const isKnowledgeElementsRestored = parseInt(await targetDatabase.runSql('SELECT  COUNT(1) FROM information_schema.tables t WHERE t.table_name = \'knowledge-elements\''));
+      expect(isKnowledgeElementsRestored).to.equal(1);
+
+      const indexCount = parseInt(await targetDatabase.runSql('SELECT COUNT(1) FROM pg_indexes ndx WHERE ndx.indexname = \'users_createdAt_idx\''));
+      expect(indexCount).to.equal(1);
+    });
+
+  });
+
+  describe('should import Airtable data', () => {
+
+    it('should import areas ', async () => {
+      // then
+      const result = parseInt(await targetDatabase.runSql('SELECT COUNT(*) FROM areas'));
```

Comment:
Pour reduire le "bruit":

```suggestion

    async function getCountFromTable(tableName) {
     return parseInt(await targetDatabase.runSql(`SELECT COUNT(*) FROM ${tableName}`));
    }
        const result = await getCountFromTable('areas')
```

Reply:
Done